src/topology/unit_interval.lean

```diff
@@ -50,7 +50,7 @@ end
 
 instance has_zero : has_zero I := ⟨⟨0, zero_mem⟩⟩
```

Comment:
What happens if you just remove this instance?

Reply:
With the import I added, everything works fine.

Reply:
That's great, thanks.

Reply:
I've moved most of the material from `topology/unit_interval` to `set/intervals/instances`.